src/lockProcessing.ts

```diff
@@ -0,0 +1,109 @@
+import { BigInt, Bytes, Address } from "@graphprotocol/graph-ts";
+import { PublicLock } from "../generated/templates/PublicLock/PublicLock";
+import { PublicLock as PublicLock7 } from "../generated/templates/PublicLock7/PublicLock";
+import { Lock, LockManager } from "../generated/schema";
+import { PublicLock as PublicLockTemplate } from "../generated/templates";
+import { PublicLock7 as PublicLockTemplate7 } from "../generated/templates";
+import { NewLock } from "../generated/Contract/Contract";
+
+export function processNewLock(event: NewLock): void {
+  let lockAddress = event.params.newLockAddress;
+  let chainPublicLock = PublicLock.bind(lockAddress);
+  let lock = bootstrapLock(chainPublicLock);
+
+  if (lock.version >= BigInt.fromI32(7)) {
+    processV7(event, lock);
+  } else {
+    processPreV7(event, lock);
+  }
+}
+
+
+function processV7(event: NewLock, lock: Lock): void {
+  let chainPublicLock = PublicLock7.bind(event.params.newLockAddress);
+  lock.address = event.params.newLockAddress;
+  lock.price = chainPublicLock.keyPrice();
+  lock.expirationDuration = chainPublicLock.expirationDuration();
+  lock.creationBlock = event.block.number;
+  lock.owner = event.params.lockOwner;
```

Comment:
👍 so that means in the future we still have a `owner` field, but it just is set to the creator for now.
I beieve we should probably mark it as deprecated?

Reply:
Initially I added the directive to mark it as deprecated in the schema but the service doesnt appear to support it https://www.apollographql.com/docs/graphql-tools/schema-directives/